### ISSUE TITLE
Reduce number of HTTP requests we make to the k8s API

### ISSUE THREAD

**Issue by yuvipanda** [jupyterhub/kubespawner#28]
When the hub starts, it poll to find the state of all user pods. However, it does this by making an individual call per pod, and this causes problems with tornado's HTTP client on large (>1k) installations. You end up with a lot of:

```
tornado.httpclient.HTTPError: HTTP 599: Timeout in request queue
```

exceptions.

We should instead move to a model similar to what kubernetes itself does:

1. Fetch all pods and put it in a local datastructure
2. Use native watch functionality to watch for pod status changes, and update local datastructure
3. Treat local datastructure as canonical source.

This would need to be properly threaded and perhaps use the kubernetes API client library.

**Comment by yuvipanda** [comment]
Should be fixed for real with #63

**Comment by minrk** [comment]
Fixed by #63
